tree/dataframe/inc/ROOT/RCsvDS.hxx

```diff
@@ -75,7 +77,8 @@ protected:
    std::string AsString();
 
 public:
-   RCsvDS(std::string_view fileName, bool readHeaders = true, char delimiter = ',', Long64_t linesChunkSize = -1LL);
+   RCsvDS(std::string_view fileName, bool readHeaders = true, char delimiter = ',', Long64_t linesChunkSize = -1LL,
+          std::string fillValue = "NaN", unsigned int lookDownForType = 10U);
```

Comment:
About `fillValue`: this would be the equivalent of pandas `na_values`, but instead of a list it is just a single value, correct? In that case, could we change the name to `naValue` ?

Reply:
`NaN` is really trying to get closer to the behaviour of pandas. Any concrete naming would do the same job. In fact, what my current functionality does is the following:

If the user have not specified what is the `fillValue`, I have a checker that `fillValue` is `NaN` (default), and in this case, it will put a  `quiet_NaN` at this position. Which as discussed in the other comment is not ideal, because works only for doubles. So from user perspective, having this variable as `NaN`, but not anything else has no impact, on how pleasent their ouput is (except for strings). They will see whatever `quiet_NaN` returned for their nummerical type.

Reply:
What would be a valid value for `fillValue`, other than `NaN`? Given that it's a default value that it would be applied to every column.

Reply:
Well, you can enter "0", "0.", "-1", "3.14", "hello", etc. Any other value different than "NaN" will be converted to the column type it belongs to. If it reamains "NaN", there is an explicit check that will cast the value to `quiet_NaN` of the corresponding column type.

Reply:
But if I specify `fillValue="hello"` and I have a column with missing values that is of type integer, what happens?

Reply:
Well, this is the same case as if currently somebody gave me a column which contains integers (internally Long Long) in the beginning, and then at some place in the column there is `"hello"`. There is no exception for that now, but I can write it. There is attempt to do `stoll("hello")` [here](https://github.com/ikabadzhov/root/blob/d3fc0f20e595662caee4e178c0f3e30efeff9c54/tree/dataframe/src/RCsvDS.cxx#L131). Since `stoll` fails it will give: `Error in <TRint::HandleTermInput()>: std::invalid_argument caught: stoll`.

I can put `stoll` in try-except block. In fact, for the first line, there are regexes to determine the column types. I can apply these checks before converting each cell: If the current cell value do not match the type if was supposed to be, I can exit the program in a controlled way telling the user what went wrong.

Reply:
I see, thanks. In the end, I guess what I'm trying to say is that having an option `fillValue` that applies to all columns might not be that useful, since columns can have different types and it can't be impossible to specify a `fillValue` that works for all of them. Also, note that pandas does not have such an option, since the `na_values` option of pandas has a different meaning: the list of things that can appear in a CSV and should be interpreted as NA.

Reply:
Makes sense. So in the initial version of our "empty cell csv reader", we can also avoid argument `fillValue` as well.